dataplane/unittest/vrf_lpm_common.h

```diff
@@ -0,0 +1,363 @@
+inline std::string AddressStr(const uint32_t& address)
+{
+	return common::ipv4_address_t(address).toString();
+}
+
+template<typename PrefixType, typename PrefixTree>
+bool FillPrefixTree(const std::vector<PrefixType>& prefixes, PrefixTree& tree, tVrfId vrf_number)
+{
+	uint32_t value = 0;
+	stats_t stats = {0, 0};
+	size_t prefix_number = prefixes.size();
+	for (tVrfId vrf = 0; vrf < vrf_number; vrf++)
+	{
+		for (size_t index = 0; index < prefix_number; index += 1 + 5 * vrf)
+		{
+			auto address = prefixes[index].address;
+			if (tree.Insert(stats, vrf, IpAddressToInternal(address), prefixes[index].mask, value) != eResult::success)
+			{
+				YANET_LOG_ERROR("Error add prefix, vrf=%d, prefix=%s/%d, value=%d\n",
+				                vrf,
+				                AddressStr(IpAddressToInternal(prefixes[index].address)).c_str(),
+				                (uint16_t)prefixes[index].mask,
+				                value);
+				return false;
+			}
+			value++;
+		}
+	}
+
+	return true;
+}
+
+template<typename PrefixTree1, typename PrefixTree2>
+bool CompareTrees(const PrefixTree1& tree1, const PrefixTree2& tree2)
+{
+	auto list1 = tree1.GetFullList();
+	auto list2 = tree2.GetFullList();
+	std::sort(list1.begin(), list1.end());
+	std::sort(list2.begin(), list2.end());
+	auto iter1 = list1.begin();
+	auto iter2 = list2.begin();
+
+	while ((iter1 != list1.end()) && (iter2 != list2.end()))
+	{
+		const auto& [vrf1, address1, mask1, value1] = *iter1;
+		const auto& [vrf2, address2, mask2, value2] = *iter2;
+		if ((vrf1 != vrf2) || (address1 != address2) || (mask1 != mask2) || (value1 != value2))
+		{
+			YANET_LOG_ERROR("[vrf=%d, %s/%d, value=%d] != [vrf=%d, %s/%d, value=%d]\n",
+			                vrf1,
+			                AddressStr(address1).c_str(),
+			                uint16_t(mask1),
+			                value1,
+			                vrf2,
+			                AddressStr(address2).c_str(),
+			                uint16_t(mask2),
+			                value2);
+			return false;
+		}
+		++iter1;
+		++iter2;
+	}
+	if (iter1 != list1.end())
+	{
+		const auto& [vrf1, address1, mask1, value1] = *iter1;
+		YANET_LOG_ERROR("list2 empty, in list1: vrf=%d, %s/%d, value=%d\n",
+		                vrf1,
+		                AddressStr(address1).c_str(),
+		                uint16_t(mask1),
+		                value1);
+		return false;
+	}
+	if (iter2 != list2.end())
+	{
+		const auto& [vrf2, address2, mask2, value2] = *iter2;
+		YANET_LOG_ERROR("list1 empty, in list2: vrf=%d, %s/%d, value=%d\n",
+		                vrf2,
+		                AddressStr(address2).c_str(),
+		                uint16_t(mask2),
+		                value2);
+		return false;
+	}
+	return true;
+}
+
+inline ipv6_address_t LastAddress(const ipv6_prefix_t& prefix)
+{
+	if (prefix.mask == 64)
+	{
+		return prefix.address;
+	}
+
+	uint64_t hi = rte_cpu_to_be_64(*(reinterpret_cast<const uint64_t*>(prefix.address.bytes)));
+	uint64_t low = rte_cpu_to_be_64(*(reinterpret_cast<const uint64_t*>(prefix.address.bytes + 8)));
```

Comment:
use memcpy when dealing with simple u64 types instead of dereferencing a pointer with reintepret casts, much cleaner this way:
```cpp
        uint64_t hi = 0, low = 0;

        std::memcpy(&hi, prefix.address.bytes, sizeof(hi));
        std::memcpy(&low, prefix.address.bytes + 8, sizeof(low));
        
        hi = rte_cpu_to_be_64(hi);
        low = rte_cpu_to_be_64(low);
```